base/linalg/dense.jl

```diff
@@ -74,10 +71,8 @@ julia> isposdef(A)
 true
 ```
 """
-function isposdef(A::AbstractMatrix{T}) where T
-    S = typeof(sqrt(one(T)))
-    isposdef!(S == T ? copy(A) : convert(AbstractMatrix{S}, A))
-end
+isposdef(A::AbstractMatrix{T}) where {T} =
+    isposdef!(copy_oftype(A, promote_type(typeof(chol(one(T))),Float32)))
```

Comment:
Copied from here: https://github.com/JuliaLang/julia/blob/2d8c0f9160812dba14c08b3602d907f561e843fe/base/linalg/cholesky.jl#L279-L280
Essentially to be able to call `cholfact` inplace.

The alternative would be to define:
`isposdef(A::AbstractMatrix) = ishermitian(A) && isposdef(cholfact(A))`
that might actually be cleaner, and avoids the copy if `A` is not Hermitian.
Thoughts?